programs/exchange_api/src/exchange_processor.rs

```diff
@@ -771,19 +853,19 @@ mod test {
                 src_account: to_src,
                 dst_account: to_dst
             },
-            ExchangeProcessor::deserialize_trade(&to_trade_account.data[..]).unwrap()
+            ExchangeProcessor::deserialize_trade(&mut keyed_accounts, 0).unwrap()
```

Comment:
Zooming in on a concrete example, I think this PR makes this code quite a bit worse than the original code. This code should read:

```rust
ExchangeState::deserialize(&to_trade_account.data)
```

Reply:
Tests got uglier but processor itself got prettier.  I could change the serialize and deserialize signatures back and still name the indexes.  I changed it because it was annoying having to flay out the keyed account members to get to the data member each time the processor wanted to serialize and deserialize.

Reply:
So prettier by what definition? In my opinion, any time you pass a function more data than it uses, it's uglier, even though calling that function took less keystrokes. Classic example: using globals.  Your passing `keyed_accounts` to most every function and then having each one reach in and deserialize just KeyedAccount::Account::data isn't too far off from putting `keyed_accounts` in a global variable.

Reply:
heh, I did consider putting `keyed_accounts` into self ;-). I agree, passing more then the function wants sucks but so does flaying out the members each time you want to call the function.  what we maybe need is a keyed_account function that returns the data given an index